Sources/Datadog/RUM/RUMMonitor/Scopes/RUMApplicationScope.swift

```diff
@@ -36,31 +48,55 @@ internal class RUMApplicationScope: RUMScope, RUMContextProvider {
     // MARK: - RUMScope
 
     func process(command: RUMCommand, context: DatadogContext, writer: Writer) -> Bool {
-        if sessionScope == nil {
+        if sessionScopes.isEmpty && !applicationActive {
             startInitialSession(on: command, context: context, writer: writer)
         }
 
-        if let currentSession = sessionScope {
-            sessionScope = sessionScope?.scope(byPropagating: command, context: context, writer: writer)
+        if activeSession == nil && command.isUserInteraction {
+            // No active sessions, start a new one
+            startNewSession(on: command, context: context, writer: writer)
+        }
+
+        if command is RUMStopSessionCommand {
+            // Reach in and grab the last active view
+            lastActiveView = activeSession?.viewScopes.first(where: { $0.isActiveView } )
```

Comment:
I'm interested in thoughts on this approach. This feels like it's pulling in some extra dependencies in the ViewScope, but I couldn't find a good way to make activating the previous SessionScope's view work, even with the existing refresh logic.